gensim/test/test_csparse.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2010 Radim Rehurek <radimrehurek@seznam.cz>
+# Licensed under the GNU LGPL v2.1 - http://www.gnu.org/licenses/lgpl.html
+from gensim.csparse.psparse import pmultiply
+
+from numpy.testing import assert_array_equal
+from numpy import asfortranarray
+from numpy.random import normal
+from functools import partial
+from scipy.sparse import rand
+from os import cpu_count
+from timeit import Timer
+import unittest
+import logging
+import scipy
+
+class TestCSparse(unittest.TestCase):
+    def testPmultiply(self):
+
+        def matmul(X, Y): return X @ Y
+        def pmatmul(X, Y): return pmultiply(X, Y)
+
+        X_rows, X_cols, factors = 10000, 100000, 10000
+
+        X = rand(X_rows, X_cols, format="csc")
+        Y = asfortranarray(normal(0.0, 1.0, (X.shape[1], factors)))
+
+        cpus = cpu_count()
+        
+        if cpus > 1:
+            t_matmul = Timer(partial(matmul, X, Y)).timeit(1)
+            t_pmatmul = Timer(partial(pmatmul, X, Y)).timeit(1)
+
+            assert(t_pmatmul < t_matmul)
+
+        assert_array_equal(matmul(X, Y), pmatmul(X, Y))
+
+        print(f'''cpus: {cpus}
```

Comment:
f-strings from 3.6, please use `format` instead.

Reply:
also, this is good as the benchmark (allows to control performance), but printing in tests isn't a good idea, please remove this when all other stuff will be ready to merge.